packages/jsActions/nanoflow-actions-native/src/permissions/RequestGenericPermission.ts

```diff
@@ -0,0 +1,65 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import { Alert, Platform } from "react-native";
+import { check, request, RESULTS, openSettings, PERMISSIONS, Permission } from "react-native-permissions";
+import { ANDROIDPermissionName, IOSPermissionName } from "../../typings/RequestGenericPermission";
+
+// BEGIN EXTRA CODE
+
+function handleBlockedPermission(): void {
+    Alert.alert("title", "need to go to settings for activation", [
+        // draft text
+        { text: "go to settings", onPress: () => openSettings() },
+        { text: "cancel" }
+    ]);
+}
+
+function mapPermissionName(permissionName: string): Permission | undefined {
```

Comment:
What about simplifying this to 

```
return PERMISSIONS[Platform.OS.toUpperCase()][permissionName.substring(0, permissionName.indexOf("_"))]
```

Reply:
As discussed, it's fine the way it is since it works and typescript narrows types implicitly. If in future someone can figure out how to simplify to oneline while typing properly, great. It's not worth stalling on IMO.